addons/loyalty/models/loyalty_reward.py

```diff
@@ -234,7 +234,7 @@ def write(self, vals):
             if vals['active']:
                 self.reward_product_id.action_unarchive()
             else:
-                self.reward_product_id.action_archive()
+                self.discount_line_product_id.action_archive()
```

Comment:
Could you also update line 235 to reflect this change ?

Reply:
Done :ok_hand: :smile: